tools/src/test/java/org/apache/kafka/trogdor/common/ExpectedTasks.java

```diff
@@ -142,17 +144,23 @@ public ExpectedTasks addTask(ExpectedTask task) {
     }
 
     public ExpectedTasks waitFor(final CoordinatorClient client) throws InterruptedException {
-        TestUtils.waitForCondition(new TestCondition() {
-            @Override
-            public boolean conditionMet() {
-                TasksResponse tasks = null;
-                try {
-                    tasks = client.tasks(new TasksRequest(null, 0, 0, 0, 0, Optional.empty()));
-                } catch (Exception e) {
-                    log.info("Unable to get coordinator tasks", e);
-                    throw new RuntimeException(e);
-                }
-                StringBuilder errors = new StringBuilder();
+        waitFor(client, DEFAULT_MAX_WAIT_MS);
+        return this;
+    }
+
+    public ExpectedTasks waitFor(final CoordinatorClient client, long maxWaitMs) throws InterruptedException {
+        TestUtils.waitForCondition(() -> {
+            TasksResponse tasks;
+            try {
+                tasks = client.tasks(new TasksRequest(null, 0, 0, 0, 0, Optional.empty()));
+            } catch (Exception e) {
+                log.info("Unable to get coordinator tasks", e);
+                throw new RuntimeException(e);
+            }
+            StringBuilder errors = new StringBuilder();
+            if (tasks.tasks().size() < expected.size()) {
+                errors.append("We expected {} tasks but received {}", tasks.tasks().size(), expected.size());
```

Comment:
I don't think this is necessary, and it makes the error log messages less useful.  Formerly we got specific details about what was missing; now, we just get a blanket error message that there aren't enough tasks.